sql/analyze.go

```diff
@@ -419,41 +457,41 @@ func simplifyOneAndExpr(left, right parser.Expr) (parser.Expr, parser.Expr) {
 					Operator: parser.GT,
 					Left:     lcmp.Left,
 					Right:    lcmp.Right,
-				}, nil
+				}, nil, true
 			}
 			// x != y
-			return left, nil
+			return left, nil, cmp == +1
```

Comment:
The `Datum.Compare` method returns `-1`, `0` or `1`. In this file I frequently use `+1` to distinguish from `-1`.